deploy.md

```diff
@@ -1,59 +1,142 @@
-Here are the instructions to deploy Activity Stream:
+# Overview
 
-#  Have the correct credentials
-# Update code and packages
+At the end of each milestone, we will start a new release process that consists of following tasks.
+
+- creating the `release` branch. A new git branch `release` will be created off the `master` to encapsulate the release preparations, for instance, building release candidates, QA check-ups, and documentation updates etc.
+
+  Once we create this branch and push it to the central repository, the release enters the code-freeze stage, any functionality that isn't already in there is postponed until the next release cycle.
+
+- building release candidate. As soon as the `release` branch lands in the central repo, the CI automatically builds the addon and ships it to the `pre-release` channel, which is supposed to be used by all the stakeholders (QA/PM/Engineers) during the release.
+
+  As a side note, there are three channels for Activity Stream.
+  * [dev](https://s3.amazonaws.com/moz-activity-streams-dev/dist/latest.html)
+  * [pre-release](https://s3.amazonaws.com/moz-activity-streams-prerelease/dist/latest.html)
+  * [release](https://moz-activity-streams.s3.amazonaws.com/dist/latest.html)
+
+- release signing-off. Upon the first build of release candidate, the release engineer sends out the release preview email to all the stakeholders. In addition, she/he files a release issue on Github to track the status through an attached checklist. The release deploy won't be initiated unless all the items get checked by the stakeholders.
```

Comment:
Minor nit: It's "GitHub" (uppercase "H"). I think there are a few other references in this file as well.